pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -343,11 +467,23 @@ func (rd *restoreDataProcessor) openSSTs(
 	if err != nil {
 		return err
 	}
-	return sendIter(iter, dirs)
+
+	return sendIter(iter, dirs, sstCh, iterAllocs, true)
 }
 
-func (rd *restoreDataProcessor) runRestoreWorkers(ctx context.Context, ssts chan mergedSST) error {
+func (rd *restoreDataProcessor) runRestoreWorkers(
+	ctx context.Context, ssts chan mergedSST, workerSSTChannels []chan mergedSST,
+) error {
 	return ctxgroup.GroupWorkers(ctx, rd.numWorkers, func(ctx context.Context, worker int) error {
+		if err := rd.mem.Grow(ctx, minWorkerMemReservation); err != nil {
+			if worker != 0 {
+				return nil // no memory to run this worker
+			}
+			return errors.Wrap(err, "insufficient memory available to run restore")
```

Comment:
what's the goal here? all workers except for the first one will silently exit, letting worker 0 continue running, but that worker 0 will not pick up work from the queues that other workers listen on, right?

Reply:
Oh, I missed this: do we only pull work off of a specific queue? I imagined fewer workers would just mean draining a common queue slower and missed that we have per-worker queues outside the worker?

Reply:
Yeah I missed this too. I made the change so that the number of worker-specific queues match the final number of workers, PTAL.